customers/schema/mutations.py

```diff
@@ -1,110 +1,167 @@
+    except BoatCertificate.DoesNotExist as e:
+        raise VenepaikkaGraphQLError(e)
+
+
+class BoatCertificateInput(graphene.InputObjectType):
     file = Upload()
     certificate_type = BoatCertificateTypeEnum()
     valid_until = graphene.Date()
     checked_at = graphene.Date()
     checked_by = graphene.String()
 
 
-class CreateBoatCertificateMutation(graphene.ClientIDMutation):
-    class Input(BoatCertificateInput):
-        boat_id = graphene.ID(required=True)
-        certificate_type = BoatCertificateTypeEnum(required=True)
+class BoatInput:
+    boat_type_id = graphene.ID()
+    registration_number = graphene.String()
+    name = graphene.String()
+    model = graphene.String()
+    length = graphene.Decimal()
+    width = graphene.Decimal()
+    draught = graphene.Decimal()
+    weight = graphene.Int()
+    propulsion = graphene.String()
+    hull_material = graphene.String()
+    intended_use = graphene.String()
+    add_boat_certificates = graphene.List(BoatCertificateInput)
+
 
-    boat_certificate = graphene.Field(BoatCertificateNode)
+class CreateBoatMutation(graphene.ClientIDMutation):
+    class Input(BoatInput):
+        owner_id = graphene.ID(required=True)
+        boat_type_id = graphene.ID(required=True)
+        length = graphene.Decimal(required=True)
+        width = graphene.Decimal(required=True)
+
+    boat = graphene.Field(BoatNode)
 
     @classmethod
-    @add_permission_required(BoatCertificate)
+    @add_permission_required(Boat, BoatCertificate)
     @transaction.atomic
     def mutate_and_get_payload(cls, root, info, **input):
-        boat = get_node_from_global_id(
-            info, input.pop("boat_id"), only_type=BoatNode, nullable=False
+        owner = get_node_from_global_id(
+            info, input.pop("owner_id"), only_type=ProfileNode, nullable=False
         )
-        input["boat"] = boat
+        input["owner"] = owner
+        try:
+            input["boat_type"] = BoatType.objects.get(id=input.pop("boat_type_id"))
+        except BoatType.DoesNotExist as e:
+            raise VenepaikkaGraphQLError(e)
+
+        add_certificates = input.pop("add_boat_certificates", [])
 
         try:
-            certificate = BoatCertificate.objects.create(**input)
+            boat = Boat.objects.create(**input)
+            add_boat_certificates(add_certificates, boat)
         except ValidationError as e:
             # Flatten all the error messages on a single list
             errors = sum(e.message_dict.values(), [])
             raise VenepaikkaGraphQLError(errors)
 
-        return CreateBoatCertificateMutation(boat_certificate=certificate)
+        return CreateBoatMutation(boat=boat)
 
 
-class UpdateBoatCertificateMutation(graphene.ClientIDMutation):
-    class Input(BoatCertificateInput):
+class UpdateBoatMutation(graphene.ClientIDMutation):
+    class Input(BoatInput):
         id = graphene.ID(required=True)
+        remove_boat_certificates = graphene.List(graphene.ID)
```

Comment:
we also need to cover the `update_boat_certificates` use case. when updating, `certificate_type` is non-required, but when adding certs, it should be required. so, might have to split `BoatCertificateInput` in 2: `CreateBoatCertificateInput` and `UpdateBoatCertificateInput` (this latter one could extend the former one and just set `required=True` for the `certificate_type`. just a quick suggestion, there might be other ways to implement this.